packages/typespec-ts/src/modular/buildCodeModel.ts

```diff
@@ -1231,17 +1231,32 @@ function applyEncoding(
 
 function mergeFormatAndEncoding(
   format: string | undefined,
-  encoding: string,
+  encoding: string | undefined,
```

Comment:
Do we need any test case for this?

Reply:
currently we can't, but we could include test cases in https://github.com/Azure/autorest.typescript/issues/2676.

the case could be prepared when encode int as string is released.